intake_xarray/raster.py

```diff
@@ -39,9 +42,17 @@ def __init__(self, urlpath, chunks, concat_dim, xarray_kwargs=None,
         super(RasterIOSource, self).__init__(metadata=metadata)
 
     def _open_files(self, files):
-        return xr.concat([xr.open_rasterio(f,chunks=self.chunks, **self._kwargs)
-                    for f in files], dim=self.dim)
+        das = []
+        for f in files:
+            da = xr.open_rasterio(f, chunks=self.chunks, **self._kwargs)
+            dim_shape = da.sizes.get(self.dim, 1)
 
+            coords = {}
+            for k, v in rev_format(self.pattern, f).items():
+                coords[k] = xr.DataArray(np.full(dim_shape, v), dims=self.dim)
```

Comment:
Could you also set the dtype here?  In the notebook, you show dates as encoded strings, but it would be nice to have actual datetime objects.  Perhaps [datetime.strptime](https://docs.python.org/3/library/datetime.html#strftime-strptime-behavior) could be used with the format from the urlpath?